src/js/components/SchemaField.js

```diff
@@ -231,6 +233,61 @@ class SchemaField extends Component {
     );
   }
 
+  renderPlacement(errorMessage, autoFocus, props) {
+    const {
+      required,
+      name,
+      schema,
+      formData,
+      onChange,
+      onBlur,
+      onFocus
+    } = props;
+
+    // example formData: [[“@zone”,“GROUP_BY”,“3”],[“@zone”,“MAX_PER”,“2"],[“hostname “,”UNIQUE”]]”
+
+    // questions:
+    // 1. when parsing and reducing the formData, can we exactly re-use the constraints marathon JSON parser/reducer?
+    // 2. do we want to render PlacementConstraintsWidget directly here, or build a higher component (ex: SpecialFormField)?
+    // 3. call the onChange() from mozilla library (the local variable defined above) with the reduced value from 1)
+    // 4. the errors prop...figure out later
+    // 5. PlacementConstraintsFrameworkAdapter can hold all the logic above?
+
+    // replace this implementation (copied from CreateServiceFormModal) with our own
+    // will be also use Batch and Transaction or can we be less generic and manually push/splice the constraints?
+    const onAddItem = (event) => {
```

Comment:
We should move the add and remove item handler into the widget. The widget should implement the standard schema form  interface.